src/transformers/models/levit/modeling_levit.py

```diff
@@ -0,0 +1,729 @@
+    def train(self, mode=True):
+        super().train(mode)
+        if mode and self.ab:
+            self.ab = {}  # clear ab cache
+
+    def get_attention_biases(self, device):
+        if self.training:
+            return self.attention_biases[:, self.attention_bias_idxs]
+        else:
+            device_key = str(device)
+            if device_key not in self.ab:
+                self.ab[device_key] = self.attention_biases[:, self.attention_bias_idxs]
+            return self.ab[device_key]
+
+    def forward(self, hidden_state):
+        batch_size, N, channels = hidden_state.shape
+        key, value = (
+            self.kv(hidden_state)
+            .view(batch_size, N, self.num_heads, -1)
+            .split([self.key_dim, self.attention_ratio * self.key_dim], dim=3)
+        )
+        key = key.permute(0, 2, 1, 3)
+        value = value.permute(0, 2, 1, 3)
+
+        query = self.q(self.q_subsample(hidden_state))
+        query = query.view(batch_size, self.resolution_**2, self.num_heads, self.key_dim).permute(0, 2, 1, 3)
+
+        attention = query @ key.transpose(-2, -1) * self.scale + self.get_attention_biases(hidden_state.device)
+        attention = attention.softmax(dim=-1)
+        hidden_state = (attention @ value).transpose(1, 2).reshape(batch_size, -1, self.out_dim_proj)
+        hidden_state = self.projection(self.activation(hidden_state))
+        return hidden_state
+
+
+class LevitMLPBlock(nn.Module):
+    """
+    MLP Residual Block a.k.a MLP 2X
+    """
+
+    def __init__(self, input_dim, hidden_dim):
+        super().__init__()
+        self.linear_up = MLPLayerWithBN(input_dim, hidden_dim)
+        self.activation = nn.Hardswish()
+        self.linear_down = MLPLayerWithBN(hidden_dim, input_dim)
+
+    def forward(self, hidden_state):
+        hidden_state = self.linear_up(hidden_state)
+        hidden_state = self.activation(hidden_state)
+        hidden_state = self.linear_down(hidden_state)
+        return hidden_state
+
+
+class LevitResidualBlock(nn.Module):
+    """
+    Residual Block for LeViT
+    """
+
+    def __init__(self, module, drop_rate):
+        super().__init__()
+        self.module = module
+        self.drop_rate = drop_rate
+
+    def forward(self, x):
+        if self.training and self.drop_rate > 0:
+            x = (
+                x
+                + self.module(x)
+                * torch.rand(x.size(0), 1, 1, device=x.device).ge_(self.drop_rate).div(1 - self.drop_rate).detach()
+            )
+            return x
+        else:
+            x = x + self.module(x)
+            return x
+
+
+class LevitEncoder(nn.Module):
+    """
+    LeViT Encoder Module
+    """
+
+    def __init__(self, config):
+        super().__init__()
+        self.num_features = config.embed_dim[-1]
+        self.embed_dim = config.embed_dim
+
+        resolution = config.image_size // config.patch_size
+
+        self.patch_embeddings = LevitPatchEmbeddings(config)
+
+        self.blocks = []
```

Comment:
```suggestion
        self.layers = []
```
In case you're going for "layer" instead of "block", you can also rename this variable here